pkg/telemetryquerylanguage/tql/parser_test.go

```diff
@@ -407,3 +442,280 @@ func testParsePath(val *Path) (GetSetter, error) {
+					},
+					Right: []*OpAndBooleanValue{
+						{
+							Operator: "and",
+							Value: &BooleanValue{
+								ConstExpr: Booleanp(true),
+							},
+						},
+					},
+				},
+				Right: []*OpOrTerm{
+					{
+						Operator: "or",
+						Term: &Term{
+							Left: &BooleanValue{
+								ConstExpr: Booleanp(false),
+							},
+						},
+					},
+				},
+			}),
+		},
+		{
+			query: `false and (true or false)`,
+			expected: set_name_test(&BooleanExpression{
+				Left: &Term{
+					Left: &BooleanValue{
+						ConstExpr: Booleanp(false),
+					},
+					Right: []*OpAndBooleanValue{
+						{
+							Operator: "and",
+							Value: &BooleanValue{
+								SubExpr: &BooleanExpression{
+									Left: &Term{
+										Left: &BooleanValue{
+											ConstExpr: Booleanp(true),
+										},
+									},
+									Right: []*OpOrTerm{
+										{
+											Operator: "or",
+											Term: &Term{
+												Left: &BooleanValue{
+													ConstExpr: Booleanp(false),
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}),
+		},
+		{
+			query: `name != "foo" and name != "bar"`,
+			expected: set_name_test(&BooleanExpression{
+				Left: &Term{
+					Left: &BooleanValue{
+						Comparison: &Comparison{
+							Left: Value{
+								Path: &Path{
+									Fields: []Field{
+										{
+											Name: "name",
+										},
+									},
+								},
+							},
+							Op: "!=",
+							Right: Value{
+								String: tqltest.Strp("foo"),
+							},
+						},
+					},
+					Right: []*OpAndBooleanValue{
+						{
+							Operator: "and",
+							Value: &BooleanValue{
+								Comparison: &Comparison{
+									Left: Value{
+										Path: &Path{
+											Fields: []Field{
+												{
+													Name: "name",
+												},
+											},
+										},
+									},
+									Op: "!=",
+									Right: Value{
+										String: tqltest.Strp("bar"),
+									},
+								},
+							},
+						},
+					},
+				},
+			}),
+		},
+		{
+			query: `name == "foo" or name == "bar"`,
+			expected: set_name_test(&BooleanExpression{
+				Left: &Term{
+					Left: &BooleanValue{
+						Comparison: &Comparison{
+							Left: Value{
+								Path: &Path{
+									Fields: []Field{
+										{
+											Name: "name",
+										},
+									},
+								},
+							},
+							Op: "==",
+							Right: Value{
+								String: tqltest.Strp("foo"),
+							},
+						},
+					},
+				},
+				Right: []*OpOrTerm{
+					{
+						Operator: "or",
+						Term: &Term{
+							Left: &BooleanValue{
+								Comparison: &Comparison{
+									Left: Value{
+										Path: &Path{
+											Fields: []Field{
+												{
+													Name: "name",
+												},
+											},
+										},
+									},
+									Op: "==",
+									Right: Value{
+										String: tqltest.Strp("bar"),
+									},
+								},
+							},
+						},
+					},
+				},
+			}),
+		},
+		// MORE TESTS TO DO
```

Comment:
Lets add these tests.  I didn't see any that used parentheses.

Reply:
There is one test that does check that parens re-adjust the grouping correctly with `or`. I've added another that verifies that parens around an `and` don't change the semantics. 

The remainder of the tests I had put in that list are duplicative of the tests in the Test_parse function, which is why I didn't implement them. They wouldn't give us any more coverage.